### ISSUE TITLE
pyproject.toml support for tools like poetry

### ISSUE THREAD

**Issue by RobRoseKnows** [indygreg/PyOxidizer#93]
Separate issue from #82 as recommended. pyproject.toml is what's [used by poetry](https://poetry.eustace.io/docs/pyproject/) and is the standard behind Pipfile.

**Comment by nathan-hoad** [comment]
@indygreg once this is done, will it mean that [flit](https://github.com/takluyver/flit) is supported as well?

**Comment by dballweg** [comment]
Running into the issue with flit not being able to pip install inside pyoxidizer.  I was able to get the pip install command that is generated to complete on my machine by putting `PYTHONPATH=/usr/local/lib/python3.7/site-packages` in front of the pip command.  Is it possible to get the modified version of distutils to link setuptools from the site-packages so that flit can build?

**Comment by svandragt** [comment]
Until poetry is installed, my  workflow is along the lines of:
```
poetry install
poetry export -f requirements.txt > src/requirements.txt
cd src
pyoxidizer run
```

After enabling the requirements.txt example inside the src/*.bzl file.

**Comment by jacobsvante** [comment]
Is there any way to run `poetry export` whenever `pyoxidizer build` is called? It's easy to miss when adding new dependencies to poetry.

**Comment by marmig0404** [comment]
> Or perhaps poetry export could be called whenever poetry dependencies change. 

could use the poetry export pre-commit hook, maybe...
